spark/sql/session/sparksession.go

```diff
@@ -41,6 +42,7 @@ func NewSessionBuilder() *SparkSessionBuilder {
 
 type SparkSessionBuilder struct {
 	connectionString string
+	channelBuilder   channel.Builder
```

Comment:
I am not quite sure that I understand this design. I think folks would expect you to build a fresh value whenever the `Build` function is called. It doesn't seem logical to mutate an existing one

Reply:
I don't think I agree. The overall behavior of the `Builder` interaction is that you can call `Build` to get a fresh `grpc.Conn`, but otherwise, I don't think there are any expectations around whether the builder needs to be fresh or not.

Reply:
I think I got a but confused over the two types of builder. I get it now